sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/EncryptedBlockBlobAsyncClient.java

```diff
@@ -0,0 +1,416 @@
+     * the <a href="https://docs.microsoft.com/rest/api/storageservices/put-block-list">Azure Docs for Put Block List</a>.
+     * <p>
+     * The data passed need not support multiple subscriptions/be replayable as is required in other upload methods when
+     * retries are enabled, and the length of the data need not be known in advance. Therefore, this method should
+     * support uploading any arbitrary data source, including network streams. This behavior is possible because this
+     * method will perform some internal buffering as configured by the blockSize and numBuffers parameters, so while
+     * this method may offer additional convenience, it will not be as performant as other options, which should be
+     * preferred when possible.
+     * <p>
+     * Typically, the greater the number of buffers used, the greater the possible parallelism when transferring the
+     * data. Larger buffers means we will have to stage fewer blocks and therefore require fewer IO operations. The
+     * trade-offs between these values are context-dependent, so some experimentation may be required to optimize inputs
+     * for a given scenario.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.cryptography.EncryptedBlockBlobAsyncClient.uploadWithResponse#Flux-int-int-BlobHTTPHeaders-Metadata-AccessTier-BlobAccessConditions}
+     *
+     * @param data The data to write to the blob. Unlike other upload methods, this method does not require that the
+     * {@code Flux} be replayable. In other words, it does not have to support multiple subscribers and is not expected
+     * to produce the same values across subscriptions.
+     * @param blockSize The size of each block that will be staged. This value also determines the size that each buffer
+     * used by this method will be and determines the number of requests that need to be made. The amount of memory
+     * consumed by this method may be up to blockSize * numBuffers. If block size is large, this method will make fewer
+     * network calls, but each individual call will send more data and will therefore take longer.
+     * @param numBuffers The maximum number of buffers this method should allocate. Must be at least two. Typically, the
+     * larger the number of buffers, the more parallel, and thus faster, the upload portion of this operation will be.
+     * The amount of memory consumed by this method may be up to blockSize * numBuffers.
+     * @param headers {@link BlobHTTPHeaders}
+     * @param metadata {@link Metadata}
+     * @param tier {@link AccessTier} for the destination blob.
+     * @param accessConditions {@link BlobAccessConditions}
+     * @return A reactive response containing the information of the uploaded block blob.
+     */
+    public Mono<Response<BlockBlobItem>> uploadWithResponse(Flux<ByteBuffer> data, int blockSize, int numBuffers,
```

Comment:
>uploadWithResponse [](start = 41, length = 18)

I think we should be able to get rid of this and just encrypt the data then pass through to the regular async client's version. That would also let us get rid of the internal version of stageBlock and commitBlockList